bin/sidekiq_wrapper_base.rb

```diff
@@ -0,0 +1,27 @@
+require 'optparse'
+
+RAILS_ROOT = File.expand_path('../..', __FILE__)
+SIDEKIQ_EXECUTABLE = File.join(RAILS_ROOT, 'bin/sidekiq')
+SIDEKIQ_BASE_ARGUMENTS = %W(#{SIDEKIQ_EXECUTABLE}
+                            --environment #{ENV['RAILS_ENV'] || 'production'}
+                            --timeout 15)
+
+# Parse options
+def parse_options(filename)
+  options = {}
+  OptionParser.new do |opts|
+    opts.banner = "Usage: #{filename} [options]"
+
+    opts.on('-wMANDATORY', '--workdir=MANDATORY', 'Set work directory') do |w|
```

Comment:
making _options_ mandatory is almost in any case bad design. Use CDW if not given.